### PR TITLE
pre-commit: add hook `check-yaml`

### PR DESCRIPTION
**Pull request by jbampton**
https://github.com/pre-commit/pre-commit-hooks?tab=readme-ov-file#check-yaml

### Description

This PR adds another check or test to our pre-commit framework.

From the official pre-commit organization.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [X] build/CI

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### How Has This Been Tested?

Ran locally:

`pre-commit run --all-files`

### CODE REVIEW

File: .pre-commit-config.yaml
Comment (DaanHoogland) [inline .pre-commit-config.yaml:38]: why this exclusion @jbampton ? are the yamls there not valid?

File: plugins/integrations/kubernetes-service/src/main/resources/conf/k8s-node.yml
Comment (jbampton) [inline plugins/integrations/kubernetes-service/src/main/resources/conf/k8s-node.yml:25]:
```suggestion
      {{ k8s.ssh.pub.key }}
```

File: .pre-commit-config.yaml
Comment (jbampton) [reply .pre-commit-config.yaml]: I removed the exclusion and did a fix up of the YAML.

**Comment by jbampton** [comment]
@blueorangutan package

**Review by DaanHoogland — APPROVED** [review]

**Review by DaanHoogland — APPROVED** [review]
clgtm

**Comment by vishesh92** [comment]
@blueorangutan test

**Review by vishesh92 — APPROVED** [review]
clgtm. can be merged once the tests passes.

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by jbampton** [comment]
@blueorangutan package

**Comment by jbampton** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Comment by nvazquez** [comment]
Hi guys @jbampton @DaanHoogland @vishesh92 I'm not able to get CKS clusters into Running state after these changes (at least after merging main branch on my PR #9102):
image

Accessing one of the CKS nodes I could check there were syntax errors which I could work around by reverting this PR changes. I haven't tried on the current main branch yet but I suspect the same issue could be present there

**Comment by DaanHoogland** [comment]
Sorry @nvazquez , I was a bit eager on this one, let's revert. cc @vishesh92 
@jbampton @nvazquez , do you see a possibility to make this check work (by options on the check or a syntax change in the yamls)?

**Pull request merged**
